Refactor to ease definition display-type specific colors

eigengrau opened the pull request:
This refactors the theme so that the primary color table can specify alternative
colors to be used in the terminal. This is meant to be of use when Emacs doesn’t
do a sufficiently good job at mapping the theme colors to the 256-XTerm palette.

eigengrau commented:
So, I’m not sure whether we could do a much better job than Emacs at mapping the theme to the XTerm palette, but I figured we might give it a shot. This PR doesn’t change any of the colors though, so the only tty-specific color is `bg1` for now.

zenorocha commented:
What do you think @film42?

film42 commented:
I think I'm a bit unqualified to review this one, but making things more customizable seems like a good idea. There are one or two other small changes that we should probably get in before merging this one since a rebase will be required.

eigengrau committed:
Refactor to ease definition display-type specific colors

This refactors the theme so that the primary color table can specify alternative
colors to be used in the terminal. This is meant to be of use when Emacs doesn’t
do a sufficiently good job at mapping the theme colors to the 256-XTerm palette.

eigengrau commented:
Rebased.

film42 commented on dracula-theme.el:
```diff
@@ -17,411 +17,420 @@
+               (rpm-spec-package-face :foreground ,rainbow-3)
+               (rpm-spec-section-face :foreground ,rainbow-7)
+               (rpm-spec-tag-face :foreground ,rainbow-2)
+               (rpm-spec-var-face :foreground "#a0522d")
```
Would you mind giving this custom color a label and moving it up?

eigengrau replied:
@film42 These face specs are unchanged from the current git HEAD, just moved around. We can change the PR into a generic refactoring PR if you prefer, though. You should be able to push to my `refactor` branch if you find anything you want to address in addition to tty-specific colors.

I’ll commit the changes you’ve suggested so far.

film42 commented on dracula-theme.el:
```diff
@@ -17,411 +17,420 @@
+               (powerline-evil-motion-face :inherit powerline-evil-base-face :background ,rainbow-3)
+               (powerline-evil-normal-face :inherit powerline-evil-base-face :background ,rainbow-6)
+               (powerline-evil-operator-face :inherit powerline-evil-base-face :background ,rainbow-4)
+               (powerline-evil-replace-face :inherit powerline-evil-base-face :background "#ff5555")
```
Would you mind giving this custom color a label and moving it up?

eigengrau committed:
Use labels for hex-coded colors

eigengrau commented:
I noticed one of the hex-coded colors already had a label. For the other I’ve added `rainbow-10`, though I’m not sure it’s prudent, since it may encourage folks to use that color in future changes and it’s not such a great fit with the rest of the scheme. Not sure why it was originally chosen.

film42 commented:
Thanks @eigengrau ! 

Agree it's kind of a weird color, but since it's already in place might be best to leave it for now and we can re-evaluate later. At least it's now easier to change.
